src/main/java/com/awesomeorg/airlineservice/controller/TicketController.java

```diff
@@ -1,14 +1,68 @@
 package com.awesomeorg.airlineservice.controller;
 
+import com.awesomeorg.airlineservice.entity.Ticket;
+import com.awesomeorg.airlineservice.protocol.CreateTicketRequest;
+import com.awesomeorg.airlineservice.protocol.UpdateTicketRequest;
 import com.awesomeorg.airlineservice.service.TicketService;
+import io.swagger.v3.oas.annotations.tags.Tag;
+import jakarta.validation.Valid;
 import lombok.RequiredArgsConstructor;
-import org.springframework.web.bind.annotation.RequestMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
 
+import java.util.List;
+
+@Tag(name = "Ticket controller", description = "Public endpoint related to tickets")
 @RestController
 @RequestMapping("/tickets")
 @RequiredArgsConstructor
 public class TicketController {
 
+
     private final TicketService ticketService;
-}
+
+
+    @GetMapping()
+    public ResponseEntity<Page<Ticket>> findTicket(@Valid final CreateTicketRequest query,
+                                                   @RequestParam(defaultValue = "0", required = false) final int pageNumber,
+                                                   @RequestParam(defaultValue = "25", required = false) final int pageSize) {
+        Page<Ticket> tickets = ticketService.findFreeTicket(query, PageRequest.of(pageNumber, pageSize));
+        return ResponseEntity
+                .status(HttpStatus.OK)
+                .body(tickets);
+    }
+
+
+    @PostMapping()
+    public ResponseEntity<Ticket> createTickets(@Valid @RequestBody final CreateTicketRequest request) {
+        List<Ticket> createdTickets = ticketService.createTickets(request);
+
+        if (!createdTickets.isEmpty()) {
+            Ticket createdTicket = createdTickets.get(0);
```

Comment:
If your service method didn't create ticket, then it should have already thrown an exception, no need to check it here